src/main/scala/edu/ie3/powerFactory2psdm/converter/GridConverter.scala

```diff
@@ -8,52 +8,50 @@ package edu.ie3.powerFactory2psdm.converter
 
 import edu.ie3.datamodel.models.input.NodeInput
 import edu.ie3.powerFactory2psdm.model.Subnet
-import edu.ie3.powerFactory2psdm.model.powerfactory.PowerFactoryGrid.Nodes
 import edu.ie3.powerFactory2psdm.model.powerfactory.{
-  PowerFactoryGrid,
-  PowerFactoryGridMaps
+  GridModel,
+  Node,
+  RawGridModel
 }
-import edu.ie3.powerFactory2psdm.util.GridPreparator
-
-import java.util.UUID
 
 /**
   * Functionalities to transform an exported and then parsed PowerFactory grid to the PSDM.
   */
 case object GridConverter {
 
-  def convert(pfGrid: PowerFactoryGrid) = {
+  def convert(pfGrid: RawGridModel) = {
     val gridElements = convertGridElements(pfGrid)
   }
 
   /**
     * Converts the grid elements of the PowerFactory grid
     *
-    * @param rawPfGrid the raw parsed PowerFactoryGrid
+    * @param rawGrid the raw parsed PowerFactoryGrid
     */
-  def convertGridElements(rawPfGrid: PowerFactoryGrid): Unit = {
-    val pfGrid = GridPreparator.prepare(rawPfGrid)
-    val pfGridMaps = PowerFactoryGridMaps(pfGrid)
-    val graph = GridGraphBuilder.build(pfGridMaps)
-    val subnets = SubnetBuilder.buildSubnets(graph, pfGridMaps.uuid2Node)
+  def convertGridElements(rawGrid: RawGridModel): Unit = {
+    val grid = GridModel.build(rawGrid)
+    val graph =
+      GridGraphBuilder.build(grid.nodes, grid.lines ++ grid.switches)
+    val nodeId2node = grid.nodes.map(node => (node.id, node)).toMap
+    val subnets = SubnetBuilder.buildSubnets(graph, nodeId2node)
     val psdmNodes = subnets.flatMap(
-      subnet => convertNodesOfSubnet(subnet, pfGridMaps.uuid2Node)
+      subnet => convertNodesOfSubnet(subnet, nodeId2node)
     )
   }
 
   /**
     * Converts all nodes within a subnet to PSDM [[NodeInput]]
     *
     * @param subnet    the subnet with reference to all PF nodes that live within
-    * @param uuid2node map that connects uuids with the associate PF [[Nodes]]
+    * @param id2node map that connects uuids with the associate PF [[Nodes]]
     * @return list of all converted [[NodeInput]]
     */
   def convertNodesOfSubnet(
       subnet: Subnet,
-      uuid2node: Map[UUID, Nodes]
+      id2node: Map[String, Node]
   ): List[NodeInput] =
-    subnet.nodeUuids
-      .map(nodeUuid => NodeConverter.convertNode(nodeUuid, uuid2node, subnet))
+    subnet.nodeIds
+      .map(nodeId => NodeConverter.convertNode(nodeId, id2node, subnet))
```

Comment:
In conjunction with my proposal for `NodeConverter`

```suggestion
 id2node
      .filter { case (nodeId, _) => subnet.nodeIds.contains(nodeId) }.map {
      case (_, node) => NodeConverter.convertNode(node, subnet)
    }.toList
```